src/main/kotlin/uk/gov/dluhc/printapi/validator/GenerateTemporaryCertificateRequestValidator.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.dluhc.printapi.validator
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.validation.Errors
+import org.springframework.validation.Validator
+import uk.gov.dluhc.printapi.models.GenerateTemporaryCertificateRequest
+import java.time.Clock
+import java.time.LocalDate
+
+@Component
+class GenerateTemporaryCertificateRequestValidator(
+    private val clock: Clock,
+    @Value("\${api.print-api.generate-temporary-certificate.valid-on-date-maximum-advance-days}") private val maxAdvanceDays: Long
```

Comment:
I struggled long and hard to come up with a suitable property name and variable name.
What I'm trying to express is the maximum number of days after "today" that a temporary certificate's `validOnDate` field can be set. IE. how far in the future can you set a temp cert's `validOnDate`

Suggestions on better names gratefully received.....

... or indeed, do we even need it in _this_ repo and PR ?
So, the business rule for generating temp certs is that they cannot be in the past and cannot be greater than x days in the future. 
I think `print-api` should validate that you cannot set the date in the past ... but I'm not sure about the future date part of this ....
My thinking is that the UI web client doesnt call this directly. The process is that the UI calls a VCA REST API (EIP1-1278), which in turn calls this REST API
The VCA REST API will definitely need to do this validation, to validate the date is not in the past and is not greater than x days in the future.
But do we need to do the "greater than x days in the future" validation in _both_ APIs?
Doing it in both APIs means we need to define the property in both APIs, and their values will need to be the same.

I wonder if we don't need it here?
What do you think?

Reply:
`api.print-api.generate-temporary-certificate.on-date.max-calendar-days-in-future`

Reply:
> `api.print-api.generate-temporary-certificate.on-date.max-calendar-days-in-future`

Ah yes, nice. I like that @Neil-Massey . My only change would be `on-date` should be `valid-on-date` - but I like the separation of the elements, and the explicit mention of calendar days 👍

Reply:
Good question regarding the necessity to do the validation in both APIs.
Usually both public API endpoints are treated as independent API and both have the same rules.  I can see the potential for configuration discrepancy though.  If I had to choose one to know the correct validation rule I'd have it in print-api (aka Certificate API).

Reply:
Yes, agreed. Lets do the business validation here in this API (but in my next PR)
VCA will call this API, and if it fails with a date too far in the future/past then this API will return a 400, which VCA can pass straight back to the UI 👍 (or deal with it in some way if we think there is value in that)